dependencies.go

```diff
@@ -162,10 +173,9 @@ func (f *faultyFile) Write(p []byte) (int, error) {
 	f.d.failDenominator++
 	if fail || f.d.failed || f.d.failDenominator >= f.d.writeLimit {
 		f.d.failed = true
-		// Write random amount of bytes on failure
-		return f.file.Write(fastrand.Bytes(fastrand.Intn(len(p) + 1)))
+		// scramble data
+		scrambleData(p)
```

Comment:
this is actually a violation of the `io.Writer` interface spec, which says `Write must not modify the slice data, even temporarily.` I doubt that doing so causes any problems here, just a bit of trivia.

Reply:
yep thought so as well the moment I hit enter. Changed it.